v7/navstories/navstories.py

```diff
@@ -26,24 +26,146 @@
 
 from __future__ import unicode_literals
 from nikola.plugin_categories import ConfigPlugin
+from nikola import utils
+from nikola.utils import LOGGER
 
+import re
 
 class NavStories(ConfigPlugin):
     """Add all stories to the navigation bar."""
 
     name = 'navstories'
     dates = {}
 
+    conf_vars = ['TRANSLATIONS', 'NAVSTORIES_PATHS', 'NAVSTORIES_MAPPING', 'NAVIGATION_LINKS_POST_NAVSTORIES']
+
+    # Indention for each level deeper in a submenu, than the highest level in that submenu
+    # overwriiten by site.config['NAVSTORIES_SUBMENU_INDENTION'] if defined
+    navstories_submenu_indention = '* '
+
+    def map_to_menu(self, entries):
+        """
+        Map form list of pages going into menu to tuble of format as NAVIGATION_LINKS and NAVIGATION_LINKS_POST_NAVSTORIES
+        List format:
+        - List of "top level entry"
+          - List
+            - Top level Menu text, or None if auto-mapped, i.e. not in NAVSTORIES_MAPPING
+            - List of pages in the top menu entry
+              - List of:
+                - navpath: List containing navigation hieracy (permalink without langinfo (initial /en/) and without the NAVSTORIES_PATHS (e.g. /pages/))
+                - Permalink
+                - Page title
+        Example:
+        [   [   'Menu text for nav',
+                [   [['nav'], '/pages/nav/', 'nav/'],
```

Comment:
NavNode is a class that you suggests that I create?
I think NavNode should contain:
Member variables:
- hierachy
- permalink
- title
And maybe a constructor method for creating the content based on the page variable (p)